src/main/java/net/neoforged/neoforge/client/event/CalculateDetachedCameraDistanceEvent.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.event;
+
+import net.minecraft.client.Camera;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.BlockGetter;
+import net.neoforged.bus.api.Event;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.fml.LogicalSide;
+import net.neoforged.neoforge.common.NeoForge;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Fired in {@linkplain Camera#setup(BlockGetter, Entity, boolean, boolean, float) Camera#setup(BlockGetter, Entity, boolean, boolean, float)} when camera is detached before calculating the distance of the camera from the {@linkplain Camera#getEntity() camera entity} based on a hard-coded maximum and a raycast.
+ *
+ * <p>This event is not {@linkplain ICancellableEvent cancellable}, and does not {@linkplain HasResult have a result}.</p>
+ *
+ * <p>This event is fired on the {@linkplain NeoForge#EVENT_BUS main Forge event bus},
+ * only on the {@linkplain LogicalSide#CLIENT logical client}.</p>
+ */
+public class CalculateDetachedCameraDistanceEvent extends Event {
+    
+    private final Camera camera;
+    private final boolean cameraFlipped;
+    
+    private double distance = 4D; // might need to be set in constructor if vanilla ever makes this dynamic
```

Comment:
Please pass it in the constructor now, makes it less likely an update to the constant will be missed when porting.